core/src/main/scala/scalan/arrays/ArrayOps.scala

```diff
@@ -435,6 +435,17 @@ trait ArrayOpsExp extends ArrayOps with BaseExp { self: ScalanExp =>
     val length: Rep[Int] = symbols.length
   }
 
+  case class ArrayInnerJoin[K, B, C, R](xs: Exp[Array[(K, B)]], ys: Exp[Array[(K, C)]], f: Exp[((B, C)) => R])
+                                       (implicit val ordK: Ordering[K], val nK: Numeric[K], val eK: Elem[K],
+                                        val eB: Elem[B], val eC: Elem[C], val eR: Elem[R])
+    extends ArrayDef[(K, R)]
```

Comment:
Actually, I don't see a pattern here. Joins intend to work with specific paired collections - of key-value pairs. Lambdas in joins are to merge values of (generally) different types into resultant one, i.e. `(A, B) -> C`. What will be the role of key type `K` in them? Keys aren't even passed there as arguments.